feat: napalm/acid work on water with shorter duration

efrec opened the pull request:
### Work done

- Napalm and acid weapons produce damaging areas when they hit water (or lava (but not void)).
- Areas on water have half duration.
- Submarines are immune to area damage.

### Addresses issue

Napalm and acid only work on water, right now, when they hit a unit or feature instead of the water directly. This is very inconsistent. We prefer to have a consistent and reduced effect on water, instead.

WatchTheFort commented on luaui/configs/DistortionGL4WeaponsConfig.lua:
```diff
@@ -1384,7 +1384,7 @@ explosionDistortionsNames['raptor_air_bomber_acid_t2_v1_acidbomb'] = {
 }
 
 explosionDistortionsNames['raptor_land_assault_acid_t2_v1_acidspit'] = {
-	GetDistortionClass("GroundAcidExplo", "Medium", {
```
What is we change the AoE value? How will this comment be kept up to date?

Commit:
napalm/acid duration is halved on water

not because it makes a ton of sense really but because it's a compromise between the current system (napalm may/may not work on water) and a new one (it works).

Commit:
group area weapons in decal code and add a fallback

Commit:
reduce very high distortion effects on some acid weapons

these are spawned on the sea floor, not at the explosion location, so look super bad on water

Commit:
allow some fire CEGs when impacting water

Commit:
remove water splash from on-water area weapons

Commit:
handle void water

Pretty dumb scenario, but:

ttl runs out, projectile explodes, random napalm or acid area appears in the middle of a void

better just to patch it probably

Commit:
is lava water? scholars may debate

efrec committed:
Merge branch 'master' into feat/flammable-water

Commit:
catch px=>area.x, unused var

Commit:
Merge branch 'feat/flammable-water' of https://github.com/efrec/Beyond-All-Reason into feat/flammable-water

PtaQQ committed:
Merge branch 'master' into feat/flammable-water

efrec committed:
Merge branch 'master' into feat/flammable-water
